File: front_end/lib/models/payment.dart
class Payment {
  final String orderId;
  final String? paypalOrderId;
  final String status;
  final double amount;
  final String? paypalStatus;
  final String paymentMethod;
  final DateTime createdAt;

  Payment({
    required this.orderId,
    this.paypalOrderId,
    required this.status,
    required this.amount,
    required this.paymentMethod,
    this.paypalStatus,
    required this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      orderId: json['orderId'] ?? '',
      paypalOrderId: json['paypalOrderId'],
      status: json['status'] ?? '',
      amount: double.parse(json['amount'].toString()),
      paypalStatus: json['paypalStatus'],
      paymentMethod: json['paymentMethod'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: back_end/api/models/payment.js
const mongoose = require('mongoose');

const paymentSchema = new mongoose.Schema({
    orderId: {
        type: String,
        required: true,
        unique: true
    },
    amount: {
        type: Number,
        required: true
    },
    currency: {
        type: String,
        required: true,
        default: 'VNĐ'
    },
    status: {
        type: String,
        enum: ['CREATED', 'CAPTURED', 'FAILED', 'COMPLETED', 'ERROR'],
        default: 'CREATED'
    },
    paypalOrderId: String,
    paymentMethod: {
        type: String,
        enum: ['paypal', 'cod'],
        required: true
    },
    payerId: String,
    createdAt: {
        type: Date,
        default: Date.now
    },
    updatedAt: {
        type: Date,
        default: Date.now
    }
});

module.exports = mongoose.model('Payment', paymentSchema);